pandas/core/arrays/datetimes.py

```diff
@@ -97,6 +97,9 @@ def _dt_array_cmp(cls, op):
 
     def wrapper(self, other):
         meth = getattr(dtl.DatetimeLikeArrayMixin, opname)
+        # TODO: return NotImplemented for Series / Index and let pandas unbox
+        # Right now, returning NotImplemented for Index fails because we
+        # go into the index implementation, which may be a bug?
```

Comment:
We definitely should defer to DataFrame.  For Index we probably should on principal, but it doesn't matter behavior-wise.  For Series there is a question of if DTA == Series should return a Series (defer) or return an ndarray (like Index)

Reply:
DTA == Series will return a Series. These arrays are analogous to ndarrays.